### PR TITLE
fix: add React.forwardRef to fix warning

### LINKED ISSUE #45387: FloatButton has console warning in 5.10.1
### Reproduction link

none

### Steps to reproduce

<FloatButton.BackTop visibilityHeight={1000} tooltip="Back to Top" />

### What is expected?

No warnings on the console

### What is actually happening?

console has warning

I'm using strict mode and it seems like strict mode causes a warning?
I don't want warnings in strict mode

```
react-dom.development.js:86 Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of DomWrapper which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node
    at button
    at FloatButton (webpack-internal:///./node_modules/antd/es/float-button/FloatButton.js:41:18)
    at DomWrapper (webpack-internal:///./node_modules/rc-motion/es/DomWrapper.js:20:90)
    at eval (webpack-internal:///./node_modules/rc-motion/es/CSSMotion.js:51:32)
    at BackTop (webpack-internal:///./node_modules/antd/es/float-button/BackTop.js:41:18)
    at div
    at App (webpack-internal:///./client/src/page/view-all-member/index.tsx:74:69)
    at RenderedRoute (webpack-internal:///./node_modules/react-router/dist/index.js:554:5)
    at Routes (webpack-internal:///./node_modules/react-router/dist/index.js:1170:5)
    at Router (webpack-internal:///./node_modules/react-router/dist/index.js:1108:15)
    at BrowserRouter (webpack-internal:///./node_modules/react-router-dom/dist/index.js:399:5)
    at LocaleProvider (webpack-internal:///./node_modules/antd/es/locale/index.js:23:5)
    at MotionWrapper (webpack-internal:///./node_modules/antd/es/config-provider/MotionWrapper.js:16:5)
    at ProviderChildren (webpack-internal:///./node_modules/antd/es/config-provider/index.js:133:5)
    at ConfigProvider (webpack-internal:///./node_modules/antd/es/config-provider/index.js:392:54)
    at div
    at App
```

| Environment | Info |
| --- | --- |
| antd | 5.10.1 |
| React | 18.2.0 |
| System | windows 10 |
| Browser | chrome 117.0.5938.92 |

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #45387

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: add React.forwardRef to fix warning |
| 🇨🇳 Chinese | 修复 FloatButton 组件报错的问题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 2e058ae</samp>

Refactored `FloatButton` component and related components to improve ref forwarding and typing. Used `composeRef` and `React.forwardRef` to merge refs and pass them to `FloatButton`. Created `interface.ts` to store common types for `FloatButton` and its usage in `BackTop` and `FloatButtonGroup`.

### <samp>🤖 Generated by Copilot at 2e058ae</samp>

*  Refactor `FloatButton` component to accept a generic `ButtonRef` type instead of specific HTML element types ([link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-28edfe59d92b75ce5389381bb8b05943da4ec4f6adc98fda85791009ff922d4bR13),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-28edfe59d92b75ce5389381bb8b05943da4ec4f6adc98fda85791009ff922d4bL23-R24),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-28edfe59d92b75ce5389381bb8b05943da4ec4f6adc98fda85791009ff922d4bL99-R101),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-28edfe59d92b75ce5389381bb8b05943da4ec4f6adc98fda85791009ff922d4bL115-R110),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-6cd07387b985064a5529e3f2613ce6a466d006de5854d04d2e7f690f6724f797R9-R10),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-6cd07387b985064a5529e3f2613ce6a466d006de5854d04d2e7f690f6724f797L66-R69))
*  Use `composeRef` to merge external and internal refs of `FloatButton` component in `BackTop` and `FloatButtonGroup` components ([link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-cc22512798c08331a87083478119980a31a5344e769bd1726eb1cee06b2608d9L1-R17),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-cc22512798c08331a87083478119980a31a5344e769bd1726eb1cee06b2608d9L31-R40),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-cc22512798c08331a87083478119980a31a5344e769bd1726eb1cee06b2608d9L75-R81),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-e5d88d36fdd37df3e6d7bd4b14ab20798187137dd85d28acaee0a597cd9a09aeL13-R13),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-e5d88d36fdd37df3e6d7bd4b14ab20798187137dd85d28acaee0a597cd9a09aeL48-R52))
*  Wrap `BackTop` component with `React.forwardRef` to enable parent component to access `FloatButton` ref ([link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-cc22512798c08331a87083478119980a31a5344e769bd1726eb1cee06b2608d9L83-R89))
*  Remove `memo` function from default export of `BackTop` component to avoid warning in development mode ([link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-cc22512798c08331a87083478119980a31a5344e769bd1726eb1cee06b2608d9L89-R95))
*  Reorder imports in `BackTop` component and remove unused imports in `FloatButtonGroup` component ([link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-cc22512798c08331a87083478119980a31a5344e769bd1726eb1cee06b2608d9L1-R17),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-e5d88d36fdd37df3e6d7bd4b14ab20798187137dd85d28acaee0a597cd9a09aeL1-R1))
*  Create `interface.ts` file to store common types for `FloatButton` component and related components ([link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-6cd07387b985064a5529e3f2613ce6a466d006de5854d04d2e7f690f6724f797R2),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-6cd07387b985064a5529e3f2613ce6a466d006de5854d04d2e7f690f6724f797R9-R10),[link](https://github.com/ant-design/ant-design/pull/45390/files?diff=unified&w=0#diff-6cd07387b985064a5529e3f2613ce6a466d006de5854d04d2e7f690f6724f797L66-R69))

### CODE REVIEW

**Comment by afc163** [comment]
需要测试用例

File: components/float-button/interface.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import type React from 'react';
+
 import type { BadgeProps } from '../badge';
 import type { TooltipProps } from '../tooltip';
 import type BackTop from './BackTop';
 import type Group from './FloatButtonGroup';
 import type PurePanel from './PurePanel';
 
+export type ButtonRef = HTMLAnchorElement & HTMLButtonElement;
```
Comment (MadCcc) [inline components/float-button/interface.ts:9]:
```suggestion
export type FloatButtonRef = HTMLAnchorElement & HTMLButtonElement;
```

File: components/float-button/interface.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import type React from 'react';
+
 import type { BadgeProps } from '../badge';
 import type { TooltipProps } from '../tooltip';
 import type BackTop from './BackTop';
 import type Group from './FloatButtonGroup';
 import type PurePanel from './PurePanel';
 
+export type FloatButtonRef = HTMLAnchorElement & HTMLButtonElement;
```
Comment (MadCcc) [inline components/float-button/interface.ts:9]:
```suggestion
export type FloatButtonRef = {
  nativeElement: HTMLAnchorElement & HTMLButtonElement;
}
```

新增的 ref 按这个格式拓展吧

Comment (zombieJ) [reply components/float-button/interface.ts:9]: ref https://github.com/ant-design/ant-design/wiki/API-Naming-rules#reference

另外，Tooltip 类组件需要改造支持这个新的形式

**Review by afc163 — APPROVED** [review]

**Pull request merged**